From 4847fc10dfd665ee1aefce74c13d05bf9b5c9dde Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Sat, 27 Aug 2022 12:03:47 +0200
Subject: [PATCH] Update CMakeLists.txt

---
 bindings/python/CMakeLists.txt | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/bindings/python/CMakeLists.txt b/bindings/python/CMakeLists.txt
index eb110f5eb0f..b3b639f367b 100644
--- a/bindings/python/CMakeLists.txt
+++ b/bindings/python/CMakeLists.txt
@@ -23,6 +23,18 @@ set_property(
     TARGET ${target_name}
     PROPERTY SWIG_DEPENDS python.i numpy.i)
 
+
+# Fix for PyPy
+# See https://gitlab.kitware.com/cmake/cmake/-/issues/23651
+# See https://github.com/conda-forge/idyntree-feedstock/pull/44
+if(Python3_INTERPRETER_ID STREQUAL "PyPy")
+    execute_process(COMMAND ${Python3_EXECUTABLE}
+                    -c "import sysconfig; print(sysconfig.get_config_var("EXT_SUFFIX"))"
+                    OUTPUT_VARIABLE IDYNTREE_PYTHON_EXT_SUFFIX)
+    string(STRIP ${IDYNTREE_PYTHON_EXT_SUFFIX} IDYNTREE_PYTHON_EXT_SUFFIX)
+    set_property (TARGET ${target_name} PROPERTY SUFFIX "${IDYNTREE_PYTHON_EXT_SUFFIX}")
+endif()
+
 if(NOT MSVC)
     set_property(
         TARGET ${target_name}
