diff --git a/bindings/python/visualize/meshcat_visualizer.py b/bindings/python/visualize/meshcat_visualizer.py
index 1c47361282..e59098bb5b 100644
--- a/bindings/python/visualize/meshcat_visualizer.py
+++ b/bindings/python/visualize/meshcat_visualizer.py
@@ -23,7 +23,10 @@ class MeshcatVisualizer:
     """
 
     def __init__(self, zmq_url=None):
-        import meshcat
+        try:
+            import meshcat
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError("the meshcat module is not found by the idyntree MeshcatVisualizer. Install explicitly meshcat-python via mamba install meshcat-python and try again.")
 
         if zmq_url is not None:
             print("Connecting to meshcat-server at zmq_url=" + zmq_url + ".")
